### ISSUE TITLE
Rolling update every 30m

### ISSUE THREAD

**Issue by haf** [zalando/spilo#536]
I'm continuously getting "Performing rolling update" about every 30m; this causes some issues if I'm working with a specific database, and depending on how the replication is set up (I haven't verified yet) and how the failover is set up, may cause dataloss (e.g. if a client is in the middle of a transaction which hasn't completed yet and doesn't implement retries, or if the WAL is not fully applied at the moment the new instance is promoted to leader).

This first creates a new pod, and then fails over to that pod.

It is triggered with this message:

```
time="2021-01-05T18:42:52Z" level=debug msg="+  zalando-postgres-operator-rolling-update-required: true" cluster-name=app/app-analytics-db pkg=cluster
time="2021-01-05T18:42:52Z" level=info msg="reason: new statefulset containers's exporter (index 1) ports do not match the current one" cluster-name=app/app-analytics-db pkg=cluster
time="2021-01-05T18:42:52Z" level=debug msg="updating statefulset" cluster-name=app/app-analytics-db pkg=cluster
time="2021-01-05T18:42:52Z" level=debug msg="patching statefulset annotations" cluster-name=app/app-analytics-db pkg=cluster
time="2021-01-05T18:42:52Z" level=debug msg="patching statefulset annotations" cluster-name=app/app-analytics-db pkg=cluster
time="2021-01-05T18:42:52Z" level=debug msg="calling Patroni API on a pod app/app-analytics-db-0 to set the following Postgres options: map[wal_level:logical]" cluster-name=app/app-analytics-db pkg=cluster
time="2021-01-05T18:42:52Z" level=debug msg="making PATCH http request: http://10.4.7.23:8008/config" cluster-name=app/app-analytics-db pkg=cluster
time="2021-01-05T18:42:52Z" level=debug msg="performing rolling update" cluster-name=app/app-analytics-db pkg=cluster
```

The thing is that there's no port index 1 for the exporter sidecar:

From the runtime pod:

```yaml
    name: exporter
    ports:
    - containerPort: 9114
      name: http-prom
      protocol: TCP
```

And in the config:

```yaml
    ports:
    - name: http-prom
      containerPort: 9114
```

So I don't see what the operator is talking about, when it says that it needs to roll my DB...?

**Comment by CyberDem0n** [comment]
Patroni/Spilo by themself never do rolling updates. It is controlled by postgres-operator.
IIRC, there is a bug in operator. It sets annotation `zalando-postgres-operator-rolling-update-required: true` on a statefulset and on certain conditions this flag is never removed. It results is rolling upgrade every 30m :(
As a workaround you can manually remove this annotation.

**Comment by haf** [comment]
@CyberDem0n I think this is trigged when we tried to use backups. Unfortunately the access to the backups is never tested, so the operator launches pods which trigger a backup as part of their bootstrapping. When this fails (in our case with the message "not found" and "googleapis.com"), the pod never reports back successfully to the operator (despite it working otherwise).

But even if there's an annotation, if you look at the logs I pasted:

> new statefulset containers's exporter (index 1) ports 

isn't there some signal to this message? How can this message even happen if it's just a stray annotation?
